.github/workflows/preview-and-release.yml

```diff
@@ -0,0 +1,88 @@
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Maven Preview and Github Release
+
+on:
+  push:
+    branches: [ dev ]
+    paths:
+     - 'src/**'
+     - '.github/**'
+     - '!.gradle/wrapper'
+     - '!.gitignore'
+     - '!LICENSE'
+     - '!THIRD PARTY NOTICES'
+     - '!*.md'
+     - '*.gradle'
+  workflow_dispatch: 
+  
+env: 
+  PREVIEW_TASK: publishSnapshotPublicationToSonatypeSnapshotRepository
+
+jobs:
+  maven_Preview:
+    environment: 
```

Comment:
we need a condition here to filter on dev

Reply:
The workflow only gets triggered by Pushes to Dev so I don't think a filter here is necessary.

Reply:
we also have the workflow dispatch trigger which allows you to select the branch.

Reply:
What would be the scenario in which trigger this snapshot release on a branch other than dev?

Reply:
I could go to the actions tab, on this workflow, and dispatch it on any other branch available. This would trigger a relase. Hopefully this release would be blocked by the environment policy, but better safe than sorry.